src/Latte/Compiler/Compiler.php

```diff
@@ -174,7 +177,8 @@ private function buildClassBody(array $tokens): string
 		$this->output = &$output;
 		$this->inHead = true;
 		$this->macroNode = new RootNode;
-		$this->htmlNode = $this->context = $this->paramsExtraction = null;
+		$this->htmlNode = $this->context = null;
+		$this->paramsExtraction = $this->defaultParamsExtraction;
```

Comment:
Default params extraction needs to be set in the beggining so macros can extend it with type information (prepend @var annotations)